io/directory.go

```diff
@@ -316,11 +344,11 @@ func (d *BasicDirectory) GetCidBuilder() cid.Builder {
 }
 
 // SwitchToSharding returns a HAMT implementation of this directory.
-func (d *BasicDirectory) SwitchToSharding(ctx context.Context) (*HAMTDirectory, error) {
+func (d *BasicDirectory) SwitchToSharding(ctx context.Context, shardWidth int) (*HAMTDirectory, error) {
```

Comment:
This is a breaking change that will have to be propagated upstream, but seems reasonable 👍